plugins/cognito/pkg/fetch/fetch.go

```diff
@@ -106,3 +95,32 @@ func (f *Fetcher) Fetch(ctx context.Context, outputWriter, errorWriter io.Writer
 
 	return nil
 }
+
+func (f *Fetcher) fetchGroups() iter.Seq2[map[string]any, error] {
```

Comment:
You can use this pattern is so many other places here and greatly simplify the code. It helps decouple the retrieval and parsing of data from writing it to its ultimate output stream.
This is how you'll be able to get rid of all those weird functions that take multiple writers. A function should be responsible for reading input _or_ writing output but not both.